src/ucp/dt/datatype_iter.c

```diff
@@ -29,6 +29,7 @@ ucp_datatype_iter_mem_dereg_some(ucp_context_h context,
     memh_index_old = 0;
     memh_index     = 0;
     ucs_for_each_bit(md_index, dt_reg->md_map) {
+        ucs_assert(memh_index < UCP_MAX_OP_MDS);
```

Comment:
may be add the check for `memh_index_old` also
```suggestion
        ucs_assert(memh_index < UCP_MAX_OP_MDS);
        ucs_assert(memh_index_old < UCP_MAX_OP_MDS);
```

Reply:
already checked below; converted to assertv